src/main/java/com/faforever/api/map/MapService.java

```diff
@@ -343,11 +342,9 @@ private void validateScenarioLua(MapLuaAccessor mapLua, MapNameBuilder mapNameBu
     Optional<Map> existingMapOptional = mapRepository.findOneByDisplayName(displayName);
     existingMapOptional
       .ifPresent(existingMap -> {
-        Optional.ofNullable(existingMap.getAuthor())
-          .filter(existingMapAuthor -> !Objects.equals(existingMapAuthor, author))
-          .ifPresent(existingMapAuthor -> {
-            throw ApiException.of(ErrorCode.MAP_NOT_ORIGINAL_AUTHOR, existingMap.getDisplayName());
-          });
+        final Player existingMapAuthor = existingMap.getAuthor();
+        if (existingMapAuthor == null || !existingMapAuthor.equals(author))
```

Comment:
Objects.equals(existingMapAuthor, author)

Reply:
I don't want nulls to be equal just for sure

Reply:
It's also more expressively, as it indicates that existingMapAuthor can be null

Reply:
Yeah all good I had thought there was an explicit null check on author that threw an error if null coming into the function but it isn't a big deal